src/client_side_reply.cc

```diff
@@ -1048,62 +1048,52 @@ clientReplyContext::purgeDoMissPurge()
 void
 clientReplyContext::purgeDoPurgeGet(StoreEntry *newEntry)
 {
-    if (newEntry) {
-        /* Release the cached URI */
-        debugs(88, 4, "clientPurgeRequest: GET '" << newEntry->url() << "'" );
-#if USE_HTCP
-        neighborsHtcpClear(newEntry, http->request, HttpRequestMethod(Http::METHOD_GET), HTCP_CLR_PURGE);
-#endif
-        newEntry->release(true);
-        purgeStatus = Http::scOkay;
-    }
-
+    purgeEntry(newEntry, Http::METHOD_GET);
     lookingforstore = 4;
     StoreEntry::getPublicByRequestMethod(this, http->request, Http::METHOD_HEAD);
 }
 
+void
+clientReplyContext::purgeEntry(StoreEntry *entry, const Http::MethodType methodType, const char *description)
```

Comment:
> Also reduced code duplication in clientReplyContext::purgeDoPurge* methods.

Are any of these changes related to fixing bug 5057? It is difficult for me to understand, based on the diff, whether these changes do more than reducing code duplication. If they do not, I would move them into a different PR to avoid distracting us from the actual bug that may need a non-trivial code changes or non-trivial reasoning to justify trivial code changes.

Reply:
By moving all purgeStatus assignments into a dedicated method (which is called from many places) I tried to eliminate the dependence of the call order. For example, the old code assigned purgeStatus to scNotFound in purgeDoPurgeGet() - thus relying on that order. Now the logic for all callers is:
* look for the to-be-purged entry
* assign purgeStatus according to the search results
* If (after all searches) purgeStatus is still scNone, report the level-1 BUG message

Reply:
You are describing PR improvements in the purging code. I certainly agree that the overall shape of that code was improved. It is an improvement worth keeping (somewhere)! However, that is not what my question was about. I am trying to understand whether these improvements fix any use case that resulted in BUG messages in the current official code.

My (weak) understanding is that bug 5057 use case is unrelated to purging. However, it is still possible that you found and fixed a bug in the purging code while working on this PR. I am trying to understand whether that is indeed the case.